src/mlpack/methods/ann/layer/layer_types.hpp

```diff
@@ -118,7 +139,8 @@ using LayerTypes = boost::variant<
     ReinforceNormal<arma::mat, arma::mat>*,
     Select<arma::mat, arma::mat>*,
     Sequential<arma::mat, arma::mat>*,
-    VRClassReward<arma::mat, arma::mat>*
+    VRClassReward<arma::mat, arma::mat>*,
+    CustomLayers*...
```

Comment:
Hm, I am not sure on this comment, but does the code `CustomLayers*...` properly expand into a pointer type for every type in `CustomLayers`?  If you are not sure, we could add another test to `feedforward_network_test.cpp` to test with two custom layers, but if you are sure that the code above is correct, then no need to do anything.

Reply:
I just tested with a very simple class like this:
```
class ForwardVisitor : public boost::static_visitor<>
{
  private:
  arma::mat input;
  arma::mat output;

  public:
  ForwardVisitor( arma::mat& input, arma::mat& output )
  : input( input ), output( output )
  {
  }
  template < typename T >
  void operator()( T* operand )
  {
    operand->Forward( input, output );
  }
};

template < typename... Args >
using LayerTypes = boost::variant< BaseLayer< double, double, double >*, Args*... >;

template < typename... Args >
class CustomFFN
{
  public:
  template < class LayerType, class... T >
  void Add( T... args )
  {
    network.push_back( new LayerType( args... ) );
  }

  void Add( LayerTypes<> layer ) { network.push_back( layer ); }

  void apply( ForwardVisitor& v )
  {
    std::for_each( network.begin(), network.end(), boost::apply_visitor( v ) );
  }

  private:
  std::vector< LayerTypes< Args... > > network;
};
```
And then use the following to test:
```
template < typename Activation, typename InputType = double, typename OutputType = double >
class EmbeddedLayer : public BaseLayer< double >
{
  public:
  template < typename eT >
  void Forward( eT& input, eT& output )
  {
    std::cout << "EmbeddedLayer" << std::endl;
  }

  size_t model;
};

template < typename Activation, typename InputType = double, typename OutputType = double >
class NewCustomLayer
{
  public:
  template < typename eT >
  void Forward( eT& input, eT& output )
  {
    std::cout << "NewCustomLayer" << std::endl;
  }

  size_t model;
};

int main()
{
  CustomFFN< EmbeddedLayer< double >, NewCustomLayer< double > > v;

  v.Add< EmbeddedLayer< double > >();
  v.Add< NewCustomLayer< double > >();
  arma::mat input;
  arma::mat output;
  ForwardVisitor visitor ( input, output );

  v.apply ( visitor );
}
```
The output is correctly printed.

Reply:
Perfect, thanks for looking into that. :+1: